packages/ui/components/SurveysList/components/SurveyMigrateForm.tsx

```diff
@@ -0,0 +1,159 @@
+import {
+  TSurvey,
+  TSurveyMigrateFormData,
+  ZSurveyMigrateFormValidation,
+} from "@formbricks/types/surveys/types";
+import { Button } from "../../Button";
+import { FormField, FormProvider } from "../../Form";
+import { Label } from "../../Label";
+import { RadioGroup, RadioGroupItem } from "../../RadioGroup";
+import { TooltipRenderer } from "../../Tooltip";
+import { migrateSurveyToOtherEnvironmentAction } from "../actions";
+
+export const SurveyMigrateForm = ({
+  defaultProducts,
+  survey,
+  onCancel,
+  setOpen,
+  onMigrated,
+}: {
+  defaultProducts: TProduct[];
+  survey: TSurvey;
+  onCancel: () => void;
+  setOpen: (value: boolean) => void;
+  onMigrated: (surveyId: string) => void;
+}) => {
+  const [selectedEnvironmentId, setSelectedEnvironmentId] = useState<string | null>(null);
+
+  const form = useForm<TSurveyMigrateFormData>({
+    resolver: zodResolver(ZSurveyMigrateFormValidation),
+    defaultValues: {
+      products: defaultProducts.map((product) => ({
+        product: product.id,
+        environments: [],
+      })),
+    },
+  });
+
+  const formFields = useFieldArray({
+    name: "products",
+    control: form.control,
+  });
+
+  const onSubmit = async () => {
+    if (!selectedEnvironmentId) {
+      toast.error("Please select an environment to migrate the survey to");
+      return;
+    }
+    try {
+      const result = await migrateSurveyToOtherEnvironmentAction({
+        surveyId: survey.id,
+        targetEnvironmentId: selectedEnvironmentId,
+      });
+      if (result?.serverError) {
+        throw new Error("Server Error while trying to migrate the survey.");
+      }
+      toast.success("Survey migrated successfully!");
+      onMigrated(survey.id);
+    } catch (error) {
+      toast.error("Failed to migrate survey");
+    } finally {
+      setOpen(false);
+    }
+  };
+
+  return (
+    <FormProvider {...form}>
+      <form
+        onSubmit={form.handleSubmit(onSubmit)}
+        className="relative flex h-full w-full flex-col gap-8 overflow-y-auto bg-white p-4">
+        <div className="space-y-8 pb-12">
+          <RadioGroup>
+            {formFields.fields.map((field, productIndex) => {
+              const product = defaultProducts.find((product) => product.id === field.product);
+              const isDisabled = survey.type !== "link" && product?.config.channel !== survey.type;
+
+              return (
+                <div key={product?.id}>
+                  <div className="mb-4 flex flex-col gap-2">
+                    <TooltipRenderer
+                      shouldRender={isDisabled}
+                      tooltipContent={
+                        <span>
+                          This product is not compatible with the survey type. Please select a different
+                          product.
+                        </span>
+                      }>
+                      <div className="w-fit">
+                        <p className="text-base font-semibold text-slate-900">
```

Comment:
Somehow, clicking on the disabled heading is submitting the form. Can you please check and fix the same?